### PR TITLE
Enable backtrace in unit tests

### PR DESCRIPTION
**Pull request by snnn**
### Description
Google test can be built either with absl/re2 or not. This PR enables the build option so that google test framework can print out a nice stacktrace when something went wrong. It helps locate test errors in CI build pipelines. 

Also,  Google test will remove the build option and make it always ON. So sooner or later we must make this change.

### CODE REVIEW

File: cmake/external/onnxruntime_external_deps.cmake [hunk 1]
```diff
@@ -37,8 +37,12 @@ if (onnxruntime_BUILD_UNIT_TESTS)
     set(gtest_disable_pthreads ON)
   endif()
   set(INSTALL_GTEST OFF CACHE BOOL "" FORCE)
-  # Set it to ON will cause crashes in onnxruntime_test_all when onnxruntime_USE_CUDA is ON
-  set(GTEST_HAS_ABSL OFF CACHE BOOL "" FORCE)
+  if (CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    # Needs to update onnxruntime/test/xctest/xcgtest.mm
+    set(GTEST_HAS_ABSL OFF CACHE BOOL "" FORCE)
```
Comment (snnn) [inline cmake/external/onnxruntime_external_deps.cmake:42]: This will be resolved in another PR with help from the mobile team.

File: cmake/onnxruntime_unittests.cmake [hunk 2]
```diff
@@ -85,29 +94,22 @@ function(AddTest)
...
-      if (MSVC)
-        target_compile_options(${_UT_TARGET} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler /wd4505>"
-                "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:/wd4505>")
-      endif()
-    endif()
     if (MSVC)
       # warning C6326: Potential comparison of a constant with another constant.
       # Lot of such things came from gtest
-      target_compile_options(${_UT_TARGET} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler /wd6326>"
+      target_compile_options(${_UT_TARGET} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options /wd6326>"
```
Comment (snnn) [inline cmake/onnxruntime_unittests.cmake:100]: "-Xcompiler" is the alias of "--compile-options". They work the same. However, we need to add "SHELL:" in front of them.

**Review by pranavsharma — APPROVED** [review]
